db/seeds.d/90_add_permissions_from_default_roles.rb

```diff
@@ -1,12 +1,13 @@
+# frozen_string_literal: true
 default_permissions = Foreman::Plugin.find("foreman_discovery").default_roles
 
 ["Discovery Reader", "Discovery Manager"].each do |role_name|
-  role = Role.find_by_name(role_name) || next
-  default_permissions[role_name].each do |permission|
-    role.add_permissions!(permission) unless role.permission_names.include?(permission.to_sym)
+  Role.ignore_locking do
+    role = Role.unscoped.find_by_name(role_name) || next
+    default_permissions[role_name].each do |permission|
+      role.add_permissions(permission) unless role.permission_names.include?(permission.to_sym)
+    end
+    role.update_attributes :origin => "discovery", :description => "Discovery plugin built-in role"
```

Comment:
`update_attributes` already saves the role, no need to call `.save!` after. Set all attributes then save to ensure it's saved only once:

```ruby
role.add_permissions(permission)
role.origin = 'discovery'
role.description = 'discovery plugin builtin'
role.save
```